src/silo/preprocessing/metadata_info.cpp

```diff
@@ -0,0 +1,200 @@
+#include "silo/preprocessing/metadata_info.h"
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+#include <boost/algorithm/string/join.hpp>
+#include <duckdb.hpp>
+
+#include "silo/config/database_config.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+
+namespace {
+
+std::vector<std::string> extractStringListValue(
+   duckdb::MaterializedQueryResult& result,
+   size_t row,
+   size_t column
+) {
+   std::vector<std::string> return_value;
+   const duckdb::Value tmp_value = result.GetValue(column, row);
+   std::vector<duckdb::Value> child_values = duckdb::ListValue::GetChildren(tmp_value);
+   std::transform(
+      child_values.begin(),
+      child_values.end(),
+      std::back_inserter(return_value),
+      [](const duckdb::Value& value) { return value.GetValue<std::string>(); }
+   );
+   return return_value;
+}
+
+std::unordered_map<std::string, std::string> validateFieldsAgainstConfig(
+   const std::unordered_map<std::string, std::string> found_metadata_fields,
+   const silo::config::DatabaseConfig& database_config
+) {
+   std::vector<std::string> config_metadata_fields;
+   std::transform(
+      database_config.schema.metadata.begin(),
+      database_config.schema.metadata.end(),
+      std::back_inserter(config_metadata_fields),
+      [](auto metadata) { return metadata.name; }
+   );
+
+   std::unordered_map<std::string, std::string> validated_metadata_fields;
+   for (const auto& pair : found_metadata_fields) {
+      if (std::find(config_metadata_fields.begin(), config_metadata_fields.end(), pair.first)
+       != config_metadata_fields.end()) {
+         validated_metadata_fields.emplace(pair);
+      }
```

Comment:
I thought of it as a feature and not as a problem. This would be one of the only upside of providing a custom config to SILO. I would also imagine that an auto-detect mode for metadata fields and types would be desirable at some point